pkg/operator/ceph/object/objectstore.go

```diff
@@ -1000,3 +1009,46 @@ func errorOrIsNotFound(err error, msg string, args ...string) error {
 	}
 	return errors.Wrapf(err, msg, args)
 }
+
+// ShouldUpdateZoneEndpointList checks whether zone endpoint list need to be updated or not
+func ShouldUpdateZoneEndpointList(zones []zoneType, desiredEndpointList []string, zoneName string) (bool, error) {
+	if zoneName == "" {
+		return false, errors.Errorf("zone name can't be empty")
+	}
+
+	zoneExists, endpoints := findZoneEndpoints(zoneName, zones)
+	if !zoneExists ||
+		(len(desiredEndpointList) == 0 && len(endpoints) == 0) {
+		return false, nil
+	}
+
+	return !listsAreEqual(desiredEndpointList, endpoints), nil
+}
+
+func findZoneEndpoints(targetZone string, zones []zoneType) (bool, []string) {
+	for _, z := range zones {
+		if z.Name == targetZone {
+			return true, z.Endpoints
+		}
+	}
+	return false, []string{}
+}
+
+func listsAreEqual(a, b []string) bool {
```

Comment:
low-level functions like this are easy to unit test. IMO they should be so that we are certain that the low-level building blocks we use to build more complex behaviors are reliable, consistent, and well-understood.